pkg/agent/agent.go

```diff
@@ -242,14 +237,36 @@ func (a *Agent) bootstrap() error {
 		a.baseSVIDKey = key
 
 		// If we're here, we need to attest/Re-attest
-		regEntryMap, err := a.attest()
+		regEntries, err := a.attest()
 		if err != nil {
 			return err
 		}
-		err = a.FetchSVID(regEntryMap, a.BaseSVID, a.baseSVIDKey)
-		if err != nil {
-			return err
+		serverId := url.URL{
+			Scheme: "spiffe",
+			Host:   a.config.TrustDomain.Host,
+			Path:   path.Join("spiffe", "cp"),
 		}
+		cmgrConfig := &cache.MgrConfig{
+			ServerCerts:    a.serverCerts,
+			ServerSPIFFEID: serverId.String(),
+			ServerAddr:     a.config.ServerAddress.String(),
+
+			BaseSVID:       a.BaseSVID,
+			BaseSVIDKey:    a.baseSVIDKey,
+			BaseRegEntries: regEntries,
+			Logger:         a.config.Log}
+
+		a.CacheMgr, err = cache.NewManager(a.ctx, cmgrConfig)
+
+		go a.CacheMgr.Init()
+		go func() {
+			<-a.CacheMgr.Done()
+			a.config.Log.Info("Cache Update Stopped")
+			if a.CacheMgr.Err() != nil {
+				a.config.Log.Warning("error:", a.CacheMgr.Err())
```

Comment:
nit: we can skip the `error:` prefix since we're logging it as warn level. If you wanna keep it, then maybe add a space after it